.changeset/spicy-heads-exist.md

```diff
@@ -0,0 +1,5 @@
+---
+"@biomejs/biome": minor
+---
+
+The rule `useExhaustiveDependencies` now report a diagnostic when the dependency list is not an array literal.
```

Comment:
```suggestion
Fixed [#5458](https://github.com/biomejs/biome/issues/5458): The rule `useExhaustiveDependencies` now reports a diagnostic when the dependency list is not an array literal.
```

Reply:
@arendjr #5458 is a bug report for the panic and it was a regression in v2.0.0-beta. Previously, the rule didn't report anything about non-array-literal dependency lists. I added a diagnostic with a fix of the panic, but the diagnostic is not related to the bug reported in #5458.

Reply:
Ah, that's fair. Feel free to leave that off then :)